[API-1062] Add Duplicate Pentest endpoint. Update changelog. (#105)
cobalthq/cobalt-api-docs: versions/v2/content/pentests.md

Remember - you can only request the report for a pentest scoped to the organization specified in the
<code>X-Org-Token</code> header.
</aside>

## Duplicate a Pentest

```sh
curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
```

> 

Reviewer: [we align it on slack] Just to formalize, to map in which states is possible to duplicate pentest :)

Author: I did a full test locally and the only state that you **cannot duplicate** is "new". All other states are valid for duplication. I will update the documentation.

> This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.

Reviewer: I would suggest adding in the document, which Pentests states you can duplicate.

> The pentest identifier has to be a valid pentest that belong to the organization specified in the `X-Org-Token` header.

Reviewer 2: This line is a bit weird to me, not sure if not being a native speaker, but `pentest identifier has to be a valid pentest` does not sound right to me.
```suggestion
The pentest identifier has to point to a valid pentest that belongs to the organization specified in the `X-Org-Token` header.
```

Author: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

Note there is **no JSON body** required for this endpoint.

> The new pentest will be in **Draft state** and will have the same settings as the pentest provided in the identifier.

Author: not a fan of "settings". Any other ideas?

Reviewer 3: I think the right business domain word is "brief" ?

### HTTP Request

`POST https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER-HERE/duplicate`

### Response

On success, a `201` response code will be returned. A response header, `Location`, will contain the URL

> within Cobalt's API of the new duplicated pentest.

Reviewer 2: `new duplicated pentest` can we just say `new pentest` for simplicity? We are already in the duplicate context.

Author: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

<aside class="notice">

> Remember - you can only request a pentest scoped to the organization specified in the <code>X-Org-Token</code> header.

Reviewer 2: ```suggestion
Remember - you can only duplicate a pentest within the organization specified in the <code>X-Org-Token</code> header.
```
just a suggestion to simplify.

Author: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

Remember - you can only request the report for a pentest scoped to the organization specified in the
<code>X-Org-Token</code> header.
</aside>

## Duplicate a Pentest

```sh
curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
```

> This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.

Reviewer 3: I'm thinking the wording "This endpoint duplicates a pentest based on the identifier provided **and creates a new pentest**." is strange. To me, duplication implies creation of a new pentest. The "and creates a new pentest" implies two actions.

Author: Yeah. I am changing it to

This endpoint creates a duplicate pentest based on the identifier provided.

> The pentest identifier has to point to a valid pentest that belongs to the organization specified in the `X-Org-Token` header.

Reviewer 3: I think this wording could be better too "identifier has to point to". 
Perhaps something like:
"The pentest to be duplicated must exist and belong to the organization specified in the `X-Org-Token` header.

> Note there is **no JSON body** required for this endpoint.

Reviewer 3: "Request body" instead of "JSON body"? 🤔
